File: TestMVVM/app/src/main/java/com/example/testmvvm/entidades/Ruta.java
package com.example.testmvvm.entidades;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

import com.example.testmvvm.constantes.Constantes;


@Entity(tableName = Constantes.NOMBRE_TABLA_RUTA,
        //primaryKeys = {"origenId","destinoId"},
        foreignKeys = {
                @ForeignKey(entity = Lugar.class,
                        parentColumns = {"id"},
                        childColumns = {"origenId"},
                        onDelete = ForeignKey.CASCADE),
                @ForeignKey(entity = Lugar.class,
                        parentColumns = {"id"},
                        childColumns = {"destinoId"},
                        onDelete = ForeignKey.CASCADE)
        })

public class Ruta {
    @PrimaryKey(autoGenerate = true)
    public long id;

    @ColumnInfo(name = "nombre")
    public String nombre;

    @ColumnInfo(name = "origenId")
    public int origenId;

    @ColumnInfo(name = "destinoId")
    public int destinoId;

    public Ruta(int lugar1, int lugar2, String nombre) {
        this.origenId = lugar1;
        this.destinoId = lugar2;
        this.nombre = nombre;
    }

    public int getorigenValue() {
        return this.origenId;
    }
    public int getdestinoValue() {
        return this.destinoId;
    }
    public String getNombreValue() {
        return this.nombre;
    }

    public Ruta() {}

}

File: RoomTestKotlin/app/src/main/java/com/example/roomtestkotlin/modelo/entidades/Ruta.kt
package com.example.roomtestkotlin.modelo.entidades

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey
import com.example.roomtestkotlin.constantes.Constantes

@Entity(tableName = Constantes.NOMBRE_TABLA_RUTA,
    //primaryKeys = {"origenId","destinoId"},
        foreignKeys = [
            ForeignKey(entity = Lugar::class,
                parentColumns = ["id"],
                childColumns = ["origenId"],
                onDelete = CASCADE),
            ForeignKey(entity = Lugar::class,
                parentColumns = ["id"],
                childColumns = ["destinoId"],
                onDelete = CASCADE)
        ])

class Ruta {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @ColumnInfo(name = "nombre")
    var nombre: String = ""

    @ColumnInfo(name = "origenId")
    var origenId: Long = 0

    @ColumnInfo(name = "destinoId")
    var destinoId: Long = 0

}